mobile-app/app/screens/AppNavigator/screens/Balances/hooks/TokenPrice.ts

```diff
@@ -0,0 +1,115 @@
+  calculatePriceRates: (fromTokenSymbol: string, pairs: PoolPairData[], amount: string) => { aToBPrice: BigNumber, bToAPrice: BigNumber, estimated: BigNumber }
+  getSelectedPoolPairs: (tokenASymbol: string, tokenBSymbol: string) => PoolPairData[]
+  getPairAmountFromLP: (symbol: string, amount: string) => { tokenAAmount: BigNumber, tokenBAmount: BigNumber, tokenASymbol: string, tokenBSymbol: string }
+}
+
+export function useTokenPrice (): DexTokenPrice {
+  const client = useWhaleApiClient()
+  const dispatch = useDispatch()
+  const blockCount = useSelector((state: RootState) => state.block.count)
+  const pairs = useSelector((state: RootState) => state.wallet.poolpairs)
+
+  const graph: GraphProps[] = pairs.map(pair => {
+    return {
+      pairId: pair.data.id,
+      a: pair.data.tokenA.symbol,
+      b: pair.data.tokenB.symbol
+    }
+  })
+
+  useEffect(() => {
+    dispatch(fetchPoolPairs({ client }))
+  }, [blockCount])
+
+  function getPairAmountFromLP (symbol: string, amount: string): { tokenAAmount: BigNumber, tokenBAmount: BigNumber, tokenASymbol: string, tokenBSymbol: string } {
+      const pair = pairs.find(pair => pair.data.symbol === symbol)
+
+      if (pair === undefined) {
+        throw new Error(`The LP symbol is not existing: ${symbol}`)
+      }
+
+      const ratioToTotal = new BigNumber(amount).div(pair.data.totalLiquidity.token)
+      // assume defid will trim the dust values too
+      return {
+        tokenAAmount: ratioToTotal.times(pair.data.tokenA.reserve).decimalPlaces(8, BigNumber.ROUND_DOWN),
+        tokenBAmount: ratioToTotal.times(pair.data.tokenB.reserve).decimalPlaces(8, BigNumber.ROUND_DOWN),
+        tokenASymbol: pair.data.tokenA.symbol,
+        tokenBSymbol: pair.data.tokenB.symbol
+      }
+  }
+
+  function getTokenPrice (symbol: string): BigNumber {
+    // active price for walletTokens based on USDT
+    const selectedPoolPairs = getSelectedPoolPairs(symbol, 'USDT')
+    const { aToBPrice } = calculatePriceRates(symbol, selectedPoolPairs, '1')
+    return aToBPrice
+  }
+
+  function getSelectedPoolPairs (tokenASymbol: string, tokenBSymbol: string): PoolPairData[] {
+    // TODO - Handle cheapest path with N hops, currently this logic finds the shortest path
+    const { path } = findPath(graph, tokenASymbol, tokenBSymbol)
+    return path.reduce((poolPairs: PoolPairData[], token, index): PoolPairData[] => {
+      const pair = pairs.find(pair => checkIfPair({
+        a: pair.data.tokenA.symbol,
+        b: pair.data.tokenB.symbol
+      }, token, path[index + 1]))
+      if ((pair == null) || index === path.length) {
+        return poolPairs
+      }
+      return [...poolPairs, pair.data]
+    }, [])
+  }
+
+  function calculatePriceRates (fromTokenSymbol: string, pairs: PoolPairData[], amount: string): { aToBPrice: BigNumber, bToAPrice: BigNumber, estimated: BigNumber} {
+    let lastTokenBySymbol = fromTokenSymbol
```

Comment:
do we need to re-store this?

Reply:
Same explanation as above, alternatively, we can `index - 1` and add more to to retrieve previous token